vscode-extension/src/context.ts

```diff
@@ -0,0 +1,20 @@
+import {ExtensionContext, OutputChannel} from 'vscode';
+import {LanguageClient} from 'vscode-languageclient/node';
+
+// Mutable object to pass around to command handlers so they
+// can reference the current state of the extension
+//
+// Things like configuration should not be stored in this context
+// as they should be resolved contextually (workspace, file, etc).
+// This state is global to the entire extension.
+// At some point, we'll most likely have a map of Workspace => LSP Client
+// for downstream callers to use.
+//
+// We're not using the VSCode's API for this since that state must be serializable
+// as it's persisted to disk.
```

Comment:
Where does this get persisted to disk?

Reply:
i figured someone coming across this might think "Why not just use the data storage api from vscode" which is like a bag of data you can keep in the VSCode `ExtensionContext`. I just wanted to point out that we can't use that because it writes to disk.

Reply:
Oh got it. I read this backwards. Makes sense.